[quant][graph] Support for quantized::mul and quantized::mul_scalar

supriyar opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #36844 [quant][graph] Add quantized::mul_relu and quantized::mul_scalar_relu ops
* **#36818 [quant][graph] Support for quantized::mul and quantized::mul_scalar**
* #36817 [quant][graph] Update quantize_dynamic_script API to take sample model args

Summary:

Test Plan:
python test_quantize_script.py test_quantized_mul
python test_quantize_script.py test_quantized_mul_scalar

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21134438](https://our.internmc.facebook.com/intern/diff/D21134438)

supriyar committed:
[quant][graph] Support for quantized::mul and quantized::mul_scalar
Summary:

Test Plan:
python test_quantize_script.py test_quantized_mul
python test_quantize_script.py test_quantized_mul_scalar

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

supriyar committed:
Update on "[quant][graph] Support for quantized::mul and quantized::mul_scalar"


Summary:

Test Plan:
python test_quantize_script.py test_quantized_mul
python test_quantize_script.py test_quantized_mul_scalar

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

supriyar committed:
Update on "[quant][graph] Support for quantized::mul and quantized::mul_scalar"


Summary:

Test Plan:
python test_quantize_script.py test_quantized_mul
python test_quantize_script.py test_quantized_mul_scalar

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 commented on torch/csrc/jit/passes/quantization.cpp:
```diff
@@ -207,10 +209,15 @@ bool alwaysRaisesException(Block* block) {
-          n->kind() == Symbol::aten("add_")) &&
-      n->input(0)->type()->isSubtypeOf(TensorType::get()) &&
+bool hasScalarInput(Node* n) {
+  bool result = false;
```
nit: better name for `result`

jerryzh168 commented on torch/csrc/jit/passes/quantization.cpp:
```diff
@@ -207,10 +209,15 @@ bool alwaysRaisesException(Block* block) {
+  bool result = false;
+  std::unordered_set<std::string> scalar_op{"add", "add_", "mul", "mul_"};
+  for (const auto& op : scalar_op) {
+    if (n->kind() == Symbol::aten(op)) {
+      result |= true;
+    }
```
this can be `result |= n->kind() == Symbol::aten(op)`

jerryzh168 replied:
actually, I think it's better to do find: https://github.com/pytorch/pytorch/blob/master/torch/csrc/jit/passes/quantization.cpp#L179

jerryzh168 replied:
you can refactor this part to a separate function and call it here: https://github.com/pytorch/pytorch/blob/master/torch/csrc/jit/passes/quantization.cpp#L171-L181

supriyar committed:
Update on "[quant][graph] Support for quantized::mul and quantized::mul_scalar"


Summary:

Test Plan:
python test_quantize_script.py test_quantized_mul
python test_quantize_script.py test_quantized_mul_scalar

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 reviewed (approved):
Looks good!
